docs/advanced/predictive-alerting.md

```diff
@@ -0,0 +1,33 @@
+# Predictive Alerting
+
+Status: **In Development**
+
+Differential can predict if a function is failing due to an unrecoverable error that requires a code change, and alert you with reproduction steps.
+
+The control-plane has a lot of context on a particular failure when it happens. It knows the function, some metadata about the source, payload, and the error message. It can use this information to predict if the error is unrecoverable or not.
```

Comment:
The control-plane has all the required context on a particular failure, when it happens.